extensions/pkg/controller/backupbucket/reconciler.go

```diff
@@ -156,12 +163,12 @@ func (r *reconciler) delete(ctx context.Context, bb *extensionsv1alpha1.BackupBu
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("failed to get backup bucket secret: %+v", err)
 	}
-	if err := extensionscontroller.DeleteFinalizer(ctx, r.client, secret, FinalizerName); err != nil {
+	if err := controllerutils.RemoveFinalizer(ctx, r.reader, r.client, secret, FinalizerName); err != nil {
```

Comment:
Why do you use the `RemoveFinalizer` function here which will retry internally while you use the `PatchFinalizers` function during `reconciliation` which won't retry?

Reply:
This is the case I tried to describe with:

> In extension controllers for example at the end of a deletion run, we should retry on conflict, because we don't know what the actuator is doing and if it would be able to shortcut another deletion request or just lead to unnecessary Cloudprovider API calls contributing to rate limits and so on.

Basically, we don't know what the actuator is doing and an entire retry of a potentially long-running reconciliation just because of the finalizer removal seems undesirable to me.

At the beginning of the reconciliation loop where the finalizer is added, the controller/actuator can definitely handle a requeue/retry, as it's more or less the first action.

Do you have any concerns with this approach or suggest to do it differently?

Reply:
OK, I get it now, but I wonder whether we could simply then also use the "retry" method at the beginning of the reconciliation. What you describe is reasonable, but I think the current code disguises these considerations and it's non-obvious if you are not into these details.

Reply:
Would some additional comments here and there also help?
Otherwise, I can also just use the retry-on-conflict variant for adding finalizers.

Reply:
Probably it would, but I think it would end up unnecessarily verbose, so just using the other function would make the code easier read- and understandable (at least for me).

Reply:
Ok, done. PTAL :)